src/plugins/hooks.js

```diff
@@ -71,6 +74,20 @@ module.exports = function(Plugins) {
 		var hookList = Plugins.loadedHooks[hook];
 		var hookType = hook.split(':')[0];
 
+		// todo: remove when breaking all hooks params by removing req/res/socket/uid
+		if (!Plugins.clsDeprecatedParamsWarnedHooks[hook]
+				&& params
+				&& Array.isArray(hookList)
+				&& hookList.length
+				&& (params.req || params.res || params.socket || params.uid)) {
+
+			Plugins.clsDeprecatedParamsWarnedHooks[hook] = true;
+
+			winston.warn('[plugins] hook `' + hook + '` \'s `params.req`, `params.res`, `params.uid` and `params.socket` are being deprecated, '
+			+ 'plugins should use the `middleware/cls` module instead to get a reference to the http-req/res and socket (which you can get the current `uid`) '
+			+ '- for more info, visit https://docs.nodebb.org/en/latest/plugins/create.html#getting-a-reference-to-req-res-socket-and-uid-within-any-plugin-hook');
```

Comment:
Also, please feel free to reword that long sentence or whatever.. i am not good at that stuff.